packages/core/scripts/collateral-umip/1_Propose.js

```diff
@@ -30,14 +30,16 @@ const proposerWallet = "0x2bAaA41d155ad8a4126184950B31F50A1513cE25";
 
 async function runExport() {
   console.log("Running Upgrade🔥");
-  console.log("Connected to network id", await web3.eth.net.getId());
+
+  let netId = await web3.eth.net.getId();
+  console.log("Connected to network id", netId);
 
   const gasEstimator = new GasEstimator(
     winston.createLogger({
       silent: true,
     }),
     60, // Time between updates.
-    100 // Default gas price.
+    netId
```

Comment:
What's the tradeoff between using network id vs chain id for stuff like this? A lot of other code (like hardhat deployments) use chain id.

Reply:
Technically [network ID is for node-node communication while chain ID is for transaction signatures](https://besu.hyperledger.org/en/stable/Concepts/NetworkID-And-ChainID/). So the Gas Estimator stores a mapping of networks to gas station API's so I think its proper in this case to use the `networkId` to specify the "network" to use.  While the [transaction helper](https://github.com/UMAprotocol/protocol/blob/master/packages/common/src/TransactionUtils.js#L18) fetches and injects the chain ID when submitting transactions.

Reply:
Hmmmm, interesting. I see what you're saying wrt how these are used in practice, but just to play devil's advocate, if we wanted to run bots on ETC, it and ETH have the same network id (because they used to be the same network!), but different chain ids. So we would be unable to differentiate when trying to estimate gas. I think the ETC/ETH split was the reason they created chainids if I remember correctly.

We're not planning to roll out to ETC anytime soon, so I don't think this is a real concern. I was mainly curious how you were thinking about it.